src/catchup/simulation/ApplyTransactionsWork.cpp

```diff
@@ -64,20 +259,33 @@ ApplyTransactionsWork::getNextLedger(
     }
 
     size_t nOps = 0;
+    auto moreOps = [&]() {
+        return mMaxOperations == 0 || (nOps < mMaxOperations);
+    };
+
     while (true)
     {
+        // sustained: mMaxOperations > 0,
+        // scaled ledger: avoid checking nOps <= mMaxOperations, mMaxOperations
+        // = 0
         while (mTransactionIter != mTransactionHistory.txSet.txs.cend() &&
-               nOps <= mMaxOperations)
+               moreOps())
```

Comment:
This doesn't seem to work the way I would expect? The old implementation was set up to guarantee that you would get between `mMaxOperations` and `mMaxOperations + 100` operations per ledger. Now I think you're forcing the operation-rate to scale with the ledger size. But they should be able to scale independently.

Reply:
how so? Old functionality is preserved, but there is a new mode where you scale ledgers with a multiplier. It depends on what `mMaxOperations` is configured to (anything above 0 will yield sustained operation-rate, while 0 just scales each ledger based on the multiplier)

Reply:
Sorry, I don't think what I wrote was a good description of what I noticed. This implementation scales the variation in the operation rate. As noted above, the old version guaranteed that it would be between `mMaxOperations` and `mMaxOperations + 100`. Your implementation reduces that guarantee to being between `mMaxOperations` and `mMaxOperations + mMultiplier * 100`. If we wanted to simulate 10x ledger size but 1x operation rate (eg. ~100M entries and ~200 ops/ledger) this would amount to a variation in the operation rate 5 times bigger than the actual desired operation rate. I don't really see that as a constant operation rate at all.

Reply:
I agree it's not ideal to scale it this way, but I'm not sure how much it matters here: the multiplier scaling is done during transaction generation, after which simulated transactions are publishes to a special archive. When we run `simulate-transactions` , we actually don't do any scaling, but replay simulated transactions from the archive with whatever ops-per-ledger , so I do not think we'd have this problem with operation rate (command flow for this is described in CommandLine.cpp right above `runGenerateOrSimulateTxs` function)